Sources/ArcGISToolkit/Components/FloorFilter/LevelSelector.swift

```diff
@@ -30,104 +33,115 @@ struct LevelSelector: View {
+        switch (isCollapsed, isTopAligned) {
+        case (true, true), (false, false): return "chevron.down.circle"
+        case (true, false), (false, true): return "chevron.up.circle"
         }
     }
     
     /// A button used to collapse the floor level list.
+    /// - Returns: The button used to collapse and expand the selector.
     @ViewBuilder func makeCollapseButton() -> some View {
         Button {
             withAnimation {
                 isCollapsed.toggle()
             }
         } label: {
-            Image(systemName: isTopAligned ? "chevron.up.circle" : "chevron.down.circle")
+            Image(systemName: iconForCollapsedState)
                 .padding(.toolkitDefault)
         }
+        .disabled(levels.count == 1)
     }
-}
-
-/// A vertical list of floor levels.
-private struct LevelsStack: View {
-    /// The view model used by the `LevelsView`.
-    @EnvironmentObject var viewModel: FloorFilterViewModel
-    
-    /// The height of the scroll view's content.
-    @State private var contentHeight: CGFloat = .zero
     
-    /// The levels to display.
-    let levels: [FloorLevel]
+    /// A button for a level in the floor level list.
+    /// - Parameter level: The level represented by the button.
+    /// - Returns: The button representing the provided level.
+    @ViewBuilder func makeLevelButton(_ level: FloorLevel) -> some View {
+        Text(level.shortName)
+            .foregroundColor(.primary)
+            .padding([.vertical], 4)
+            .frame(maxWidth: .infinity)
+            .background {
+                RoundedRectangle(cornerRadius: 5)
+                    .fill(buttonColorFor(level))
+            }
+            .onTapGesture {
+                viewModel.setLevel(level)
+                if isCollapsed && levels.count > 1 {
+                    isCollapsed.toggle()
+                }
+            }
+    }
     
-    var body: some View {
+    /// A scrollable list of buttons; one for each level to be displayed.
+    /// - Returns: The scrollable list of level buttons.
+    @ViewBuilder func makeLevelButtons() -> some View {
         ScrollViewReader { proxy in
             ScrollView {
-                VStack {
-                    ForEach(levels, id: \.id) { level in
-                        Toggle(
-                            isOn: Binding(
-                                get: {
-                                    viewModel.selectedLevel == level
-                                },
-                                set: { newIsOn in
-                                    guard newIsOn else { return }
-                                    viewModel.setLevel(level)
-                                }
-                            )
-                        ) {
-                            Text(level.shortName)
-                                .modifier(LevelNameFormat())
-                        }
-                        .toggleStyle(.selectableButton)
```

Comment:
I found that SwiftUI's default button styles seem to be opinionated about their minimum width. This was going to cause issues if users set the width below what the styles allowed. After much experimentation I found that simply using `Text` with a rounded rectangle background was the best way forward. It also allowed for the removal of this computed binding and some custom toggle styles we'd defined to make it work correctly.